src/pandantic/plugins/pandas.py

```diff
@@ -8,12 +8,12 @@
 To register plugin: `from pandantic.plugins import pandas_plugin #(or *)`
 """
 import logging
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Hashable, Iterable, Optional
 
 import pandas as pd
-from pydantic import BaseModel
+from pydantic import BaseModel, ValidationError
 
-from pandantic.basemodel import CoreValidator
+from pandantic.validators.pandas import PandasValidator
 
 
 @pd.api.extensions.register_dataframe_accessor("pydantic")
```

Comment:
So with naming, I generally try to structure the semantics to include the whole path.

Let me explain. So for a pydantic + pandas plugin `pandantic` inlcudes both (good name), but `pd.DataFrame.pandantic...` is pandas-DataFrame-pandas-pydantic. You can see what I mean? The name hold redundant info. Pandantic is teh engine that allows `pydantic` models to be applied to a `pandas.DataFrame`.

You are the lead here, I will let you have the final call, I just wanted to explain the convention I apply.

Reply:
OK, I did have a look but I might have a good solution here.

Currently, the pandas plugin only accepts the Pydantic BaseClass as it seems. The goal of pandantic is, especially with the dependency injection pattern, to have the option to use different schemas (like the @dataclass schema). 

In case the plugin method is only usable by a Pydantic schema, `pydantic` makes more sense. The same for a dataclass schema, the method would better be named `dataclass`.

In case the plugin would be able to extract the schematype and validate accordantly, the method name `pandantic` would make the most sense. 

What do you think? So as it stands right now, `pydantic` is the right choice.

Reply:
Great point! It would be annoying to copy a lot of code just make a new plugin class for each schema type.

I changed it to `pandantic`!